custom_components/tahoma/cover.py

```diff
@@ -121,9 +121,13 @@ def update(self):
             )
             
         if CORE_TARGET_CLOSURE_STATE in self.tahoma_device.active_states:
-            self._position = 100 - self.tahoma_device.active_states.get(
+            self._position = self.tahoma_device.active_states.get(
```

Comment:
Why did you remove `100 -` here, but you added the `PositionableHorizontalAwningUno` to the `TAHOMA_HORIZONTAL_DEVICES`, which adds the `100-` below? You could keep the 100 - here probably, and just not add this device type to the `TAHOMA_HORIZONTAL_DEVICES`

Reply:
The `PositionableHorizontalAwningUno` has to be of type `TAHOMA_HORIZONTAL_DEVICES`. If not, the awning moves in the opposite direction.

My problem at the moment is, that the HA displays the awning at both end positions (completely closed and completely open) as `closed` with `current_position = 0` when I change the code in the following way. But I don't know why - it looks as expected.

Is there a way to see the `core:TargetClosureState` without logging?

```
        if CORE_TARGET_CLOSURE_STATE in self.tahoma_device.active_states:
            self._position = 100 - self.tahoma_device.active_states.get(
                CORE_TARGET_CLOSURE_STATE
            )
            # Fix for unknown position after moving with remote control and closure states over 100
            if self._position < 0:
                self._position = self._position + 50

        if getattr(self, "_position", False):
            # HorizontalAwning devices need a reversed position that can not be obtained via the API
            if self.tahoma_device.widget in TAHOMA_HORIZONTAL_DEVICES:
                self._position = 100 - self._position

            # TODO Check if this offset is really necessary
            if self._position <= 5:
                self._position = 0
            if self._position >= 95:
                self._position = 100
```

Reply:
If you look at ```developer-tools/state```, you can have a look at the `current_position`. However, this is not the original value since there are some modifications in the snippet above. Best way to debug is to add logging indeed.                     

One thing that is not 100% clear for me, is why you need the part below. It was something that had to be refactored anyway, because it was a bit 'ugly'. For horizontal awning devices that needs a reversed position, the code below is subtracting the position from 100, to get back to the original value. (because in the part above it has already be subtracted at the CORE_TARGET_CLOSURE_STATE). That's why I didn't understand why you had to remove the `100 -` part. 


Sorry for the mess. Is it a bit more clear now? :D

```python
     if getattr(self, "_position", False):
            # HorizontalAwning devices need a reversed position that can not be obtained via the API
            if self.tahoma_device.widget in TAHOMA_HORIZONTAL_DEVICES:
                self._position = 100 - self._position
```

Reply:
> If you look at developer-tools/state, you can have a look at the current_position. However, this is not the original value since there are some modifications in the snippet above. Best way to debug is to add logging indeed.

Ok, than I have to do this in that way. I know the `current_position` entry in the developer section, but that's not really helpful (as you said)

> One thing that is not 100% clear for me, is why you need the part below. It was something that had to be refactored anyway, because it was a bit 'ugly'. For horizontal awning devices that needs a reversed position, the code below is subtracting the position from 100, to get back to the original value. (because in the part above it has already be subtracted at the CORE_TARGET_CLOSURE_STATE). That's why I didn't understand why you had to remove the 100 - part.

At the moment I don't know why it was necessary yesterday. I changed the code locally to the one I posted above. The problem here is (and that's a big problem), that HA set the position to `closed` for both end positions (closed and open). But I don't know why.

Maybe some more information I found out the last weeks:
- The `PositionableHorizontalAwningUno` is an extra plug in order to make normal motors to wireless motors. But this plug don't know anything about the current position. That's why these types only know something about a "target state".

- In my installation the `CORE_TARGET_CLOSURE_STATE` could be `105`. I think, that is the value when the extra plug doesn't know the state and forgot the target closure state.
So we have the problem with the initial implementation that the position is set to `-5` first and then to `105` which is then set to `100`.

I pushed the current code that runs on my installation locally at the moment (with the problem described above).

Reply:
Oh the Uno part makes sense. I thought that all `io:` devices would know the state, but now the naming of TargetClosureState makes sense indeed. 

How does the online portal show the state? Do they return unknown if it is above 100? 

`is_closed` is the function you could also have a look at, they currently decide if the cover is closed based on the position in your case.